app/service/database_dump.py

```diff
@@ -0,0 +1,118 @@
+import logging
+import re
+import subprocess
+from datetime import datetime
+from pathlib import Path
+
+from app.config import (
+    ADMIN_POSTGRES_DATABASE,
+    ADMIN_POSTGRES_HOST,
+    ADMIN_POSTGRES_PASSWORD,
+    ADMIN_POSTGRES_USER,
+)
+
+_LOGGER = logging.getLogger(__name__)
+_LOGGER.setLevel(logging.INFO)
+
+
+def validate_postgres_param(value: str) -> str:
+    """
+    Allow only alphanumeric, dots, underscores and hyphens in Postgres parameters.
+
+    :param value: The Postgres parameter to validate
+    returns str: The validated parameter
+    :raises ValueError: If the parameter is invalid
+    """
+    if value is None or not re.match(r"^[\w.-]+$", value):
+        raise ValueError(f"Invalid Postgres parameter: {value}")
+    return value
+
+
+def delete_local_file(file_path: str) -> None:
+    """Delete a local file safely, logging any issues."""
+    path = Path(file_path)
+
+    try:
+        if path.exists():
+            path.unlink()
+            _LOGGER.debug(f"Deleted local file: {file_path}")
+    except Exception as e:
+        _LOGGER.error(f"⚠️ Failed to delete file {file_path}: {e}")
+        raise
+
+
+def get_database_dump() -> str:
+    """
+    Dumps the PostgreSQL database to a local SQL file.
+
+    Generates a timestamped `.sql` file using `pg_dump` and stores it
+    in the current working directory with secure permissions.
+
+    :raises subprocess.CalledProcessError: If the `pg_dump` command fails.
+    :raises RuntimeError: If security checks fail or the operation times out.
+    :return str: The path to the generated SQL dump file.
+    """
+    timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
+    dump_file = Path(f"navigator_dump_{timestamp}.sql")
+
+    if dump_file.exists():
+        raise RuntimeError(f"Dump file already exists: {dump_file}")
+
+    # pg_dump command
+    cmd = [
+        "pg_dump",
+        "--no-password",  # Force password to come from environment only
+        "-h",
+        validate_postgres_param(ADMIN_POSTGRES_HOST),
+        "-U",
+        validate_postgres_param(ADMIN_POSTGRES_USER),
+        "-d",
+        validate_postgres_param(ADMIN_POSTGRES_DATABASE),
+        "-f",
+        str(dump_file),
+        "--no-privileges",
+        "--no-owner",
+    ]
+
+    # Set environment with password
+    env = {"PGPASSWORD": ADMIN_POSTGRES_PASSWORD}
+
+    try:
+        _LOGGER.info(f"🚀 Starting database dump to {dump_file}")
+
+        result = subprocess.run(
+            cmd,
+            check=True,
+            env=env,
+            capture_output=True,
+            text=True,
+            shell=False,
+            timeout=300,
```

Comment:
issue: magic number! what is the unit of this? assuming millisecs:
```suggestion
            timeout_ms = 300
            timeout=timeout_ms,
```